var/spack/repos/builtin/packages/cudnn/package.py

```diff
@@ -5,229 +5,172 @@
-    depends_on('cuda@10.1.0:10.1.999', when='@7.6.3.30-10.1-linux-x64')
-    depends_on('cuda@10.1.0:10.1.999', when='@7.6.3.30-10.1-linux-ppc64le')
-    depends_on('cuda@10.1.0:10.1.999', when='@7.5.0.56-10.1-linux-x64')
-    depends_on('cuda@10.1.0:10.1.999', when='@7.5.0.56-10.1-linux-ppc64le')
-
-    # CUDA 10.0
-    depends_on('cuda@10.0.0:10.0.999', when='@7.6.5.32-10.0-osx-x64')
-    depends_on('cuda@10.0.0:10.0.999', when='@7.6.5.32-10.0-linux-x64')
-    depends_on('cuda@10.0.0:10.0.999', when='@7.6.5.32-10.0-linux-ppc64le')
-    depends_on('cuda@10.0.0:10.0.999', when='@7.6.4.38-10.0-osx-x64')
-    depends_on('cuda@10.0.0:10.0.999', when='@7.6.4.38-10.0-linux-x64')
-    depends_on('cuda@10.0.0:10.0.999', when='@7.6.4.38-10.0-linux-ppc64le')
-    depends_on('cuda@10.0.0:10.0.999', when='@7.5.1.10-10.0-linux-x64')
-    depends_on('cuda@10.0.0:10.0.999', when='@7.5.1.10-10.0-linux-ppc64le')
-    depends_on('cuda@10.0.0:10.0.999', when='@7.5.0.56-10.0-linux-x64')
-    depends_on('cuda@10.0.0:10.0.999', when='@7.5.0.56-10.0-linux-ppc64le')
-
-    # CUDA 9.2
-    depends_on('cuda@9.2.0:9.2.999', when='@7.6.5.32-9.2-linux-x64')
-    depends_on('cuda@9.2.0:9.2.999', when='@7.6.5.32-9.2-linux-ppc64le')
-    depends_on('cuda@9.2.0:9.2.999', when='@7.6.4.38-9.2-linux-x64')
-    depends_on('cuda@9.2.0:9.2.999', when='@7.6.4.38-9.2-linux-ppc64le')
-
-    # CUDA 9.1
-    depends_on('cuda@9.1.0:9.1.999', when='@7.1.3-9.1-linux-x64')
-    depends_on('cuda@9.1.0:9.1.999', when='@7.1.3-9.1-linux-ppc64le')
-
-    # CUDA 9.0
-    depends_on('cuda@9.0.0:9.0.999', when='@7.6.5.32-9.0-linux-x64')
-    depends_on('cuda@9.0.0:9.0.999', when='@7.6.4.38-9.0-linux-x64')
-    depends_on('cuda@9.0.0:9.0.999', when='@7.3.0.29-9.0-linux-x64')
-    depends_on('cuda@9.0.0:9.0.999', when='@7.2.1.38-9.0-linux-x64')
-
-    # CUDA 8.0
-    depends_on('cuda@8.0.0:8.0.999', when='@6.0-8.0-linux-x64')
-    depends_on('cuda@8.0.0:8.0.999', when='@5.1-8.0-linux-x64')
+    '5.1-8.0': {
+        'Linux-x86_64': ('c10719b36f2dd6e9ddc63e3189affaa1a94d7d027e63b71c3f64d449ab0645ce')},
+
+}
+
+
+class Cudnn(Package):
+    """NVIDIA cuDNN is a GPU-accelerated library of primitives for deep
+    neural networks"""
+
+    homepage = "https://developer.nvidia.com/cudnn"
+
+    # Latest versions available at:
+    #     https://developer.nvidia.com/rdp/cudnn-download
+    # Archived versions available at:
+    #     https://developer.nvidia.com/rdp/cudnn-archive
+    # Note that download links don't work from command line,
+    # need to use modified URLs like in url_for_version.
+
+    maintainers = ['adamjstewart', 'bvanessen']
+
+    for ver, packages in _versions.items():
+        key = "{0}-{1}".format(platform.system(), platform.machine())
+        pkg = packages.get(key)
+        cudnn_ver, cuda_ver = ver.split('-')
+        long_ver = "{0}-{1}".format(cudnn_ver, cuda_ver)
+        if pkg:
+            version(long_ver, sha256=pkg[0], expand=False)
```

Comment:
```suggestion
            version(long_ver, sha256=pkg, expand=False)
```
This should fix the failing tests. In Python, `('abc')` is equivalent to `'abc'`, so you're actually indexing the first element of the string.